crates/nu-parser/src/parser.rs

```diff
@@ -1023,131 +1041,151 @@ pub fn parse_internal_call(
         working_set.enter_scope();
     }
 
+    let mut flags_ended = false;
+
     while spans_idx < spans.len() {
         let arg_span = spans[spans_idx];
 
         let starting_error_count = working_set.parse_errors.len();
-        // Check if we're on a long flag, if so, parse
-        let (long_name, arg) = parse_long_flag(working_set, spans, &mut spans_idx, &signature);
 
-        if let Some(long_name) = long_name {
-            // We found a long flag, like --bar
-            if working_set.parse_errors[starting_error_count..]
-                .iter()
-                .any(|x| matches!(x, ParseError::UnknownFlag(_, _, _, _)))
-                && signature.allows_unknown_args
-            {
-                working_set.parse_errors.truncate(starting_error_count);
-                let arg = parse_unknown_arg(working_set, arg_span, &signature);
-
-                call.add_unknown(arg);
-            } else {
-                call.add_named((long_name, None, arg));
+        if !flags_ended {
+            if parse_flag_terminator(working_set, spans, spans_idx) {
+                // Pass it along for `extern` and `def --wrapped` commands
+                if is_known_external || signature.allows_unknown_args {
+                    call.add_positional(Expression::new(
```

Comment:
shouldn't this be `call.add_unknown` if `signature.allows_unknown_args` is true?